### PR TITLE
Transformation for high-order functions

### PR DESCRIPTION
**Pull request by alnkravchenko**

### CODE REVIEW

**Review by rssh — APPROVED** [review]

**Comment by rssh** [comment]
oops,  cl

File: compiler-plugin/src/test/scala/cc/Test12.scala [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
-   */
+    try {
+      val (code, output) =
+        dotcInvocations.compileAndRunFilesInDir("testdata/set12/m1", "cpstest.Test12m1")
+      val reporter       = dotcInvocations.reporter
+      println("summary: " + reporter.summary)
+      println(s"output=${output}")
+      assert(false, "Expected exception was not thrown")
+    } catch {
+      case ex: CpsTransformException =>
```
Comment (rssh) [inline compiler-plugin/src/test/scala/cc/Test12.scala:21]: Actually we don't want to throw exception from invocation.
(it's just forgotten try/catch)
Thr correct way is to catch CpsTransformException and use report.error for diagnostic.

**Review by rssh — CHANGES_REQUESTED** [review]

File: compiler-plugin/src/test/scala/cc/Test12.scala (hunk 1 above)
Comment (alnkravchenko) [reply compiler-plugin/src/test/scala/cc/Test12.scala:21]: But the `DotcInvocations.compileAndRunFilesInDirs` method throws RuntimeException when compilation is failed

Comment (rssh) [reply compiler-plugin/src/test/scala/cc/Test12.scala:21]: Ugu...  let's think, 
  Mb create subclass of Runtime exception for compilation error and check one ?

**Review by rssh — APPROVED** [review]

**Pull request merged**